src/map/script.c

```diff
@@ -11591,14 +11591,16 @@ BUILDIN(getstatus)
 		case 3: script_pushint(st, sd->sc.data[id]->val3); break;
 		case 4: script_pushint(st, sd->sc.data[id]->val4); break;
 		case 5:
-		{
-			const struct TimerData *td = timer->get(sd->sc.data[id]->timer);
+			if (sd->sc.data[id].infinite_duration) {
```

Comment:
You need to use -> here. I don't know if you noticed that your build failed.
But actually the assertion in timer_get in timer.c is still wrong.
The only thing you would have to add here is to return infinite duration in the else case(td == NULL).
The error @Ragn0 got is because no return value is pushed to the script engine in this case.